File: haohaninfo/交易實務案例/95.R
#取得即時報價，詳細在技巧51
source("function.R")

#設定初始的資料格式
Qty <- 0
lastMinute <- 0
lastQty <- 0

while(index!=0){
 #取得成交資訊
 Mdata<-GetMatchData(DataPath,Date)
 MatchAmount <- as.numeric(Mdata[[1]][4])
 HMTime <- as.numeric(paste0(substr(Mdata[[1]][1],1,2),substr(Mdata[[1]][1],4,5)))
 
 #若為初始值，即更新最新一筆資訊
 if(lastQty==0){
  lastQty <- MatchAmount
  lastMinute <- HMTime
 }else{
  #若非初始值，即更新最新一筆資訊
  #換分鐘則新增一筆資料
  if(HMTime > lastMinute){
   lastQty <- MatchAmount
   lastMinute <- HMTime
   Qty <- 0
  }else{
   Qty <- MatchAmount - lastQty

  }
 }

 #當每分鐘累計量超過1000時出場
 if(Qty >= 1000){
  if (index ==1){
   index <- 0
   print("Cover Buy Success!")
   break
  }else if(index==(-1)){
   index <- 0
   print("Cover Sell Success!")
   break
  }

 }

}

File: haohaninfo/即時範例/95.py
# -*- coding: UTF-8 -*-

#取得報價資訊，詳情請查看技巧51
execfile('function.py')

#定義指標變數
Qty=[]
lastHMTime=""
lastAmount=0

#假設目前倉位為買方，進場部分請參考第七章
index=1
orderPrice=10300

#取得成交資訊
for i in getMatch():		
  MatchInfo=i.split(',')
  HMTime=MatchInfo[0][0:2]+MatchInfo[0][3:5]
  MatchAmount=int(MatchInfo[3])
  
  if lastAmount==0:
   lastAmount=MatchAmount
   lastHMTime=HMTime
  if HMTime==lastHMTime:
   Qty=MatchAmount-lastAmount
  else:
   Qty=0
   lastAmount=MatchAmount
   lastHMTime=HMTime 
  
  #出場判斷
  if Qty>=1000:
   if index==1:
    index=0
    print MatchInfo[0],"Order Sell Success!"
    break 
   if index==-1:
    index=0
    print MatchInfo[0],"Order Buy Success!"
    break

  print Qty